Source/EasyNetQ/RabbitBus.cs

```diff
@@ -74,20 +91,37 @@ public class RabbitBus : IBus
             return PublishAsync(message, conventions.TopicNamingConvention(typeof(T)));
         }
 
-        public virtual async Task PublishAsync<T>(T message, string topic) where T : class
+        public virtual Task PublishAsync<T>(T message, string topic) where T : class
         {
             Preconditions.CheckNotNull(message, "message");
             Preconditions.CheckNotNull(topic, "topic");
-            var messageType = typeof (T);
+
+            return PublishAsync(message, c => c.WithTopic(topic));
+        }
+
+        public virtual async Task PublishAsync<T>(T message, Action<IPublishConfiguration> configure) where T : class
+        {
+            Preconditions.CheckNotNull(message, "message");
+            Preconditions.CheckNotNull(configure, "configure");
```

Comment:
actually this will throw an exception, but if not awaited will throw an `AggregatedException` that contains an innerExpection = `ArgumentNullException`

we can create async tests, but I don't know if the version of nunit we're using supports it, can you @tkirill open a separated issue for this?